Remove the helm.Match() methods

jandubois opened the pull request:
They were added in 678b0ba, but are overly complex, given that the only usage was locating specific resource kinds in a list of nodes. The list version of `Match()` also did not actually work (which didn't
matter, as it wasn't used anywhere).

This commit provides a much simplified version of `matchNodeInlist()` and gets rid of all the `Match()` methods. This restores `helm/config.go` to 100% test coverage, which was the original motivation for this change. 😄

mook-as commented on kube/kube_test.go:
```diff
@@ -268,9 +268,9 @@ func getBasicConfig() (map[string]interface{}, error) {
 
 // matchNodeInList iterates through a list of nodes and returns the first one
 // where node.Match(target) succeeds
```
Please update the comment.

jandubois replied:
Good catch! Commit amended.

mook-as reviewed (changes requested):
Basically looks fine (I'm assuming CI will catch any issues), just need to update the doc comment.

jandubois committed:
Remove the helm.Match() methods

They were added in 678b0ba, but are overly complex, given that the
only usage was locating specific resource kinds in a list of nodes.
The list version of Match() also did not actually work (which didn't
matter, as it wasn't used anywhere).

This commit provides a much simplified version of matchNodeInlist()
and gets rid of all the Match() methods. This restores helm/config.go
to 100% test coverage, which was the original motivation for this
change. :)
